### PR TITLE
[MJDEPS-Documentation] - Added readme.md for Apache Maven MJDEPS Plugin

### PR DESCRIPTION
**Pull request by KathrynN**
Added readme for ease of new developers getting involved, primarily made up from [shared utils](https://github.com/apache/maven-shared-utils/blob/master/readme.md) and notes made from personal correspondence.  Essentially, things I will forget if I don't contribute for a month or so.

### CODE REVIEW

File: readme.md [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+--------------
+
+We accept Pull Requests via GitHub. The [developer mailing list][dev-ml-list] is the
+main channel of communication for contributors.  
+There are some guidelines which will make applying PRs easier for us:
++ Create a branch from where you want to base your work (this is usually the master branch).
+  Push your changes to a branch in your fork of the repository.
++ Make commits of logical units.
++ You can use version 9 of Java, but be aware that the code has target of version 6.
++ Respect the original code style: by using the same [codestyle][code-style],
```
Comment (khmarbaise) [inline readme.md]: Can you please check the current site of [maven-jdep-plugin](http://maven.apache.org/plugins/maven-jdeps-plugin/plugin-info.html) cause it's targeted JDK 7.

**Comment by khmarbaise** [comment]
Can you please make the case of the file according to other repos fo example https://github.com/apache/maven-help-plugin ? 
Could you please hightlight the parts which you have noted from your personal correspondence so we can make up the other repos the same way...cause you have realized there is missing something in the description...it would much greater if you could add CONTRIBUTION.md file and a .github/pull_request_template.md (you can simply copy them from the help plugin above and recheck if they are ok..? would you be so kind?

File: readme.md (hunk 1 above)
Comment (KathrynN) [reply readme.md]: My bad!

**Comment by KathrynN** [comment]
Yes of course, changed to match "MANIFEST.mf"
The parts added were around testing - I find it difficult to remember the exact Prun-its command after a week away, and certainly a new contributor would have no idea where to look
Added those files (although as far as I could see on maven-help-plugin CONTRIBUTION.md is a copy of README.md, so I may have missed something)

**Pull request merged**